drivers/intc/intc_arm_v8m_nvic.c

```diff
@@ -0,0 +1,223 @@
+		NVIC_SetPriority(spec->irq, spec->priority);
+		config->irqs[spec->irq] = irq;
+	}
+
+	return 0;
+}
+
+static int arm_v8m_nvic_enable_irq(const struct intc_irq_spec *spec)
+{
+	const struct device *intc = spec->intc;
+	struct intc_arm_v8m_nvic_data *data = intc->data;
+	const struct intc_arm_v8m_nvic_config *config = intc->config;
+
+	if (spec->irq >= config->num_irqs) {
+		return -ENODEV;
+	}
+
+	K_SPINLOCK(&data->lock) {
+		NVIC_EnableIRQ(spec->irq);
+	}
+
+	return 0;
+}
+
+static int arm_v8m_nvic_disable_irq(const struct intc_irq_spec *spec)
+{
+	const struct device *intc = spec->intc;
+	struct intc_arm_v8m_nvic_data *data = intc->data;
+	const struct intc_arm_v8m_nvic_config *config = intc->config;
+
+	if (spec->irq >= config->num_irqs) {
+		return -ENODEV;
+	}
+
+	K_SPINLOCK(&data->lock) {
+		NVIC_DisableIRQ(spec->irq);
+	}
+
+	return 0;
+}
+
+static int arm_v8m_nvic_irq_is_enabled(const struct intc_irq_spec *spec)
+{
+	const struct device *intc = spec->intc;
+	struct intc_arm_v8m_nvic_data *data = intc->data;
+	const struct intc_arm_v8m_nvic_config *config = intc->config;
+	int ret;
+
+	if (spec->irq >= config->num_irqs) {
+		return -ENODEV;
+	}
+
+	K_SPINLOCK(&data->lock) {
+		ret = NVIC_GetEnableIRQ(spec->irq);
+	}
+
+	return ret;
+}
+
+static int arm_v8m_nvic_trigger_irq(const struct intc_irq_spec *spec)
+{
+	const struct device *intc = spec->intc;
+	struct intc_arm_v8m_nvic_data *data = intc->data;
+	const struct intc_arm_v8m_nvic_config *config = intc->config;
+
+	if (spec->irq >= config->num_irqs) {
+		return -ENODEV;
+	}
+
+	K_SPINLOCK(&data->lock) {
+		NVIC_SetPendingIRQ(spec->irq);
+	}
+
+	return 0;
+}
+
+static int arm_v8m_nvic_release_irq(struct intc_irq *irq)
+{
+	const struct intc_irq_spec *spec = irq->spec;
+	const struct device *intc = spec->intc;
+	struct intc_arm_v8m_nvic_data *data = intc->data;
+	const struct intc_arm_v8m_nvic_config *config = intc->config;
+
+	if (spec->irq >= config->num_irqs) {
+		return -ENODEV;
+	}
+
+	K_SPINLOCK(&data->lock) {
+		NVIC_DisableIRQ(spec->irq);
+		config->irqs[spec->irq] = NULL;
+	}
+
+	return 0;
+}
+
+static const struct intc_driver_api arm_v8m_nvic_api = {
+	.request_irq = arm_v8m_nvic_request_irq,
+	.enable_irq = arm_v8m_nvic_enable_irq,
+	.disable_irq = arm_v8m_nvic_disable_irq,
+	.irq_enabled = arm_v8m_nvic_irq_is_enabled,
+	.trigger_irq = arm_v8m_nvic_trigger_irq,
+	.release_irq = arm_v8m_nvic_release_irq,
+};
+
+static inline uint32_t arm_v8m_nvic_get_active_irq_num(void)
+{
+	return __get_IPSR() - 16;
+}
+
+#define ARM_V8M_NVIC_DEFAULT_IRQ_HANDLER_DEFINE(inst)						\
+	INTC_DT_ARM_V8M_NVIC_DIRECT_IRQ_HANDLER_DEFINE(arm_v8m_nvic_default_handler##inst)	\
+	{											\
+		uint32_t irq_num = arm_v8m_nvic_get_active_irq_num();				\
+		struct intc_irq **irqs = arm_v8m_nvic_irqs##inst;				\
+		struct intc_irq *irq = irqs[irq_num];						\
+												\
+		if (irq != NULL) {								\
+			irq->handler(irq);							\
+		} else {									\
+			while (1) {								\
```

Comment:
Not understanding this at all.  Presumably you get here on a spurious interrupt?  We have an existing spurious interrupt handler API the arch layers all honor, as that's a pretty routine glitch during device bringup (e.g. "oops typo in the interrupt number", or "oops, the bootloader left the UART RX interrupt enabled" -- absolutely routine things).  Entering a silent infinite loop seems like exactly the wrong thing to do.